ddtrace/tracer/tracer.go

```diff
@@ -417,6 +426,9 @@ func (t *tracer) StartSpan(operationName string, options ...ddtrace.StartSpanOpt
 			span.Service = newSvc
 		}
 	}
+	if t.longrunner != nil {
+		t.longrunner.trackSpan(span)
```

Comment:
You can check if the span is sampled before tracking. This will reduce the impact significantly as we sample by default ~10 chunks/s

Reply:
Isn't the sampling decision only finalized when the trace is finished? (for things like rate and priority sampling?)

At minimum perhaps we can do something here like not tracking when: 
`span.context.trace.samplingDecision != decisionDrop`

Reply:
It's applied just before on the go tracer: https://github.com/DataDog/dd-trace-go/blob/d1291d6c9f2061fe053d8809ee12c90d208d8962/ddtrace/tracer/tracer.go#L417-L420
I think some other tracers like the python one have late evaluation though (evaluate only if there's distributed tracing headers propagation or flush)

Reply:
Hmm. I saw the `t.sample` call, but it seems that function doesn't always set `samplingDecision` which can leave it in . In the span finish function: https://github.com/DataDog/dd-trace-go/blob/d1291d6c9f2061fe053d8809ee12c90d208d8962/ddtrace/tracer/span.go#L404 
It looks like for the priority and rate based sampling the "final" decision on sampling (if it hasn't already been set) is made there.